### PR TITLE
code_chatbot_wgrounding

### PR DESCRIPTION
**Pull request by cetrarom2**
### Description

*Provide a description of your changes here. If this PR is related to an issue, list the issue number/name here*

### Assignee

*Assignees: cetrarom2*

## PR checklist
*Please ensure the following:*
- [ ] This comment contains a description of changes (with reason).
- [ ] All changes were tested.
- [ ] If you've fixed a bug mention the issue number/name.
- [ ] Apply approriate tags (e.g. documentation, bug)

### CODE REVIEW

**Review by zbyosufzai — COMMENTED** [review]
Marco can you add the images to the image folder and link them in the notebook that way?

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    "$0.80 per month based on:<br>\n",
+    "Search Enterprise Edition. Number of requests per month = 100<br>\n",
+    "Search LLM Add-On. Number of requests per month = 100<br>\n",
+    "Data Index. Amount of GiB indexed per month = 5<br>\n",
+    "\n",
+    "## Get started\n",
+    "\n",
+    "### Create a search app\n",
+    "\n",
+    "As the __[GCP_Grounding](https://github.com/STRIDES/NIHCloudLabGCP/blob/main/notebooks/GenAI/GCP_Grounding.ipynb)__ tutorial explains, we need to create an 'Application' under the 'Agent Builder' on the console. Start by searching up 'Agent Builder' on the console."
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:32]: Done

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+   "source": [
+    "![Agent_builder](../../images/1_chatbot_grounding.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a1588ce-2e85-4761-8137-98985fbc5fd1",
+   "metadata": {},
+   "source": [
+    "Once you open the 'Agent Builder,' you will see a list of all the apps available to you. Click on 'CREATE APP' to get started."
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:48]: Done

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+   "source": [
+    "![chat](../../images/12_chatbot_grounding.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01130517-3265-4b85-a4ae-076d696da20f",
+   "metadata": {},
+   "source": [
+    "The Chat selection offers three options, and you can begin using the ChatBot by editing the System instructions. To do this, click on __Edit__ in the top right corner."
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:192]: Done.

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+   "source": [
+    "![parameters](../../images/14_chatbot_grounding.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e1465d4-1ebf-46b9-8635-7c2722293eb9",
+   "metadata": {},
+   "source": [
+    "Select __'Vertex AI search'__ as the __'Grounding source'__, and in the __'Vertex AI datastore path'__ field, input the project ID, location, collections, and datastore ID in this format: __'projects/{PROJECT_ID}/locations/global/collections/default_collection/dataStores/{DATA_STORE_ID}'__. Once you've entered the required information, simply click save. You can retrieve the Project ID and Data Store ID by executing the following Python code:"
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:224]: Done.

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "124b324f-f513-447c-8f0f-fba69fbb20e4",
+   "metadata": {},
+   "source": [
+    "If you submit a prompt to the BotChat and receive an error, it's likely that the Vertex AI datastore path is incorrect. Once you have the correct path to connect the ChatBot to the datastore, you can prompt the application with questions related to the specific domain information stored in the datastore. \n",
+    "\n",
+    "While both Retrieval-Augmented Generation (RAG) and Grounding are techniques used in natural language processing (NLP), they serve different purposes and operate at different levels. While RAG enhance the quality and accuracy of text generation by incorporating relevant information from external knowledge sources, Grounding ensures that the generated text is consistent with the information in the data soource. Both techniques are crucial for building robust and reliable NLP systems. RAG helps to improve the factual accuracy and coherence of generated text, while grounding ensures that the text is consistent with the real world and avoids hallucinations.\n",
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:253]: Done.

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+    "fastqc -t 23 *.fastq.gz -o /path/to/output/dir && multiqc /path/to/output/dir -o /path/to/output/dir\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4f822e4-46ca-4b0a-bda6-01cb13dd3fe5",
+   "metadata": {},
+   "source": [
+    "![prompt02](../../images/17_chatbot_grounding.png)"
```
[the remark this answers is not in the document]
Comment (cetrarom2) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:314]: Done.

File: images/1_chatbot_grounding.png
[the remark this answers is not in the document]
Comment (cetrarom2) [reply images/1_chatbot_grounding.png:1]: Done.

File: images/13_chatbot_grounding.png
[the remark this answers is not in the document]
Comment (cetrarom2) [reply images/13_chatbot_grounding.png:1]: Done.

File: images/14_chatbot_grounding.png
[the remark this answers is not in the document]
Comment (cetrarom2) [reply images/14_chatbot_grounding.png:1]: Done.

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ea8ba437-8abe-4340-9385-1fc50bcad49d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROJECT_ID = '<PROJECT_ID>'\n",
```
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:234]: fixed this

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+    "DATA_STORE_ID = '<DATA_STORE_ID>'"
```
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:235]: fixed this

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+   "source": [
+    "![vertex_ai](../../images/15_chatbot_grounding.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "124b324f-f513-447c-8f0f-fba69fbb20e4",
+   "metadata": {},
+   "source": [
+    "If you submit a prompt to the BotChat and receive an error, it's likely that the Vertex AI datastore path is incorrect. Once you have the correct path to connect the ChatBot to the datastore, you can prompt the application with questions related to the specific domain information stored in the datastore. \n",
```
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb:251]: fixed this

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb]: fixed

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb]: fixed

File: notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (zbyosufzai) [reply notebooks/GenAI/GCP_Code_Chatbot_wGrounding.ipynb]: fixed

**Pull request merged**